File: Nian_Karl/QL/src/org/uva/ql/ast/type/IntType.java
package org.uva.ql.ast.type;

import org.uva.ql.visitor.TypeVisitor;

public class IntType extends Type {

	@Override
	public boolean isInt() {
		return true;
	}

	@Override
	public boolean isEqual(Type type) {
		return type.isInt();
	}

	@Override
	public String toString() {
		return "Int";
	}

	@Override
	public <T> T accept(TypeVisitor<T> visitor) {
		return visitor.visit(this);
	}

}

File: FelipezConde/QuestionnaireLanguage/AST/Types/IntType.cs

namespace AST.Types
{
    public class IntType : Type
    {
        public override bool IsInt()
        {
            return true;
        }

        public override bool IsEqual(Type type)
        {
            return type.IsInt();
        }

        public override T Accept<T>(Visitors.ITypeVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override string ToString()
        {
            return "int";
        }
    }
}
